src/SW_Domain.c

```diff
@@ -0,0 +1,153 @@
+                             " Please select from \"xy\" and \"s\".",
+                             MyFileName, value);
+                }
+                strcpy(SW_Domain->DomainType, value);
+                break;
+            case 1: // Number of X slots
+                SW_Domain->nDimX = atoi(value);
+                break;
+            case 2: // Number of Y slots
+                SW_Domain->nDimY = atoi(value);
+                break;
+            case 3: // Number of S slots
+                SW_Domain->nDimS = atoi(value);
+                break;
+            case 4: // Start year
+                y = atoi(value);
+
+                if (y < 0) {
+                    CloseFile(&f, LogInfo);
+                    LogError(LogInfo, LOGFATAL,
+                             "%s: Negative start year (%d)", MyFileName, y);
+                }
+                SW_Domain->startyr = yearto4digit((TimeInt) y);
+                SW_Model->addtl_yr = 0; // Could be done anywhere; SOILWAT2 runs don't need a delta year
+                break;
+            case 5: // End year
+                y = atoi(value);
+
+                if (y < 0) {
+                    CloseFile(&f, LogInfo);
+                    LogError(LogInfo, LOGFATAL,
+                             "%s: Negative ending year (%d)", MyFileName, y);
+                }
+                SW_Domain->endyr = yearto4digit((TimeInt) y);
+                break;
+            case 6: // Start day of year
+                SW_Domain->startstart = atoi(value);
+                fstartdy = swTRUE;
+                break;
+            case 7: // End day of year
+                tempdoy = atoi(value);
+                SW_Domain->endend = (tempdoy == 365) ?
+                                Time_get_lastdoy_y(SW_Domain->endyr) : 365;
+                fenddy = swTRUE;
+                break;
+            case KEY_NOT_FOUND: // Unknown key
+                LogError(LogInfo, LOGWARN, "%s: Ignoring an unknown key, %s",
+                         MyFileName, key);
+                break;
+        }
+    }
+
+	if (SW_Domain->endyr < SW_Domain->startyr) {
+		CloseFile(&f, LogInfo);
+        LogError(LogInfo, LOGFATAL, "%s: Start Year > End Year", MyFileName);
+	}
+
+    // Check if start day of year was not found
+    if(!fstartdy) {
+        LogError(LogInfo, LOGWARN, "Domain.in: Missing Start Day - using 1\n");
+        SW_Domain->startstart = 1;
+    }
+
+	SW_Domain->startstart += ((SW_Model->isnorth) ? DAYFIRST_NORTH : DAYFIRST_SOUTH) - 1;
+
+    // Check if end day of year was not found
+	if (!fenddy) {
+		SW_Domain->endend = (SW_Model->isnorth) ?
+                        Time_get_lastdoy_y(SW_Domain->endyr) : DAYLAST_SOUTH;
+        LogError(LogInfo, LOGWARN,
+                "Domain.in: Missing End Day - using \"%d\"\n", SW_Domain->endend);
+	}
+
+	SW_Model->daymid = (SW_Model->isnorth) ? DAYMID_NORTH : DAYMID_SOUTH;
+
+    // Copy information to SW_MODEL
```

Comment:
Let's create a dedicated function that copies time from SW_DOMAIN to SW_MODEL, e.g.,
```
void SW_DOM_setModelTime(SW_DOMAIN *SW_Domain, SW_MODEL *SW_Model) {
    SW_Model->startyr = SW_Domain->startyr; // Copy start year
    SW_Model->endyr = SW_Domain->endyr; // Copy end year
    SW_Model->startstart = SW_Domain->startstart; // Copy start doy
    SW_Model->endend = SW_Domain->endend; // Copy end doy
}
```

which is called by `SW_CTL_read_inputs_from_disk()` after the call to `SW_DOM_read()` -- this separates the reading of domain (which will happen once) from setting SW_Model (which happen many times)